<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Free space assignment | The portfolio of Nicky Klaver</title>
  <meta name="description" content="2 Free space assignment | The portfolio of Nicky Klaver" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Free space assignment | The portfolio of Nicky Klaver" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="rstudio/dsfb2_workflows_portfolio" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Free space assignment | The portfolio of Nicky Klaver" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cv.html"/>
<link rel="next" href="guerilla-analytics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/core-js-2.5.3/shim.min.js"></script>
<script src="libs/react-18.2.0/react.min.js"></script>
<script src="libs/react-18.2.0/react-dom.min.js"></script>
<script src="libs/reactwidget-1.0.0/react-tools.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/reactable-0.4.4/reactable.css" rel="stylesheet" />
<script src="libs/reactable-binding-0.4.4/reactable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="cv.html"><a href="cv.html"><i class="fa fa-check"></i><b>1</b> CV</a></li>
<li class="chapter" data-level="2" data-path="free-space-assignment.html"><a href="free-space-assignment.html"><i class="fa fa-check"></i><b>2</b> Free space assignment</a>
<ul>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#introduction-single-cell-rna-sequencing"><i class="fa fa-check"></i>Introduction single cell RNA sequencing</a>
<ul>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#start-of-the-free-time-project"><i class="fa fa-check"></i>Start of the free time project:</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#meaning-of-words"><i class="fa fa-check"></i>Meaning of words</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#r-packages-for-single-cell-rna-sequencing"><i class="fa fa-check"></i>R packages for Single cell RNA sequencing</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#revealing-the-umi-matrix"><i class="fa fa-check"></i>Revealing the Umi Matrix</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#analyse-the-unique-molecular-identifier-umi-matrix"><i class="fa fa-check"></i>Analyse the Unique Molecular Identifier (UMI) matrix</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#analysis-overview"><i class="fa fa-check"></i>Analysis overview</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#scrna-course"><i class="fa fa-check"></i>scRNA course!?!</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#seurat-analysis"><i class="fa fa-check"></i>Seurat analysis</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#seurat-pre-processing"><i class="fa fa-check"></i>Seurat pre-processing</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#seurat-dimensionality-reduction-and-clustering"><i class="fa fa-check"></i>Seurat dimensionality reduction and clustering</a></li>
<li class="chapter" data-level="" data-path="free-space-assignment.html"><a href="free-space-assignment.html#seurat-expression-profile"><i class="fa fa-check"></i>Seurat expression profile</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="guerilla-analytics.html"><a href="guerilla-analytics.html"><i class="fa fa-check"></i><b>3</b> Guerilla analytics</a>
<ul>
<li class="chapter" data-level="" data-path="guerilla-analytics.html"><a href="guerilla-analytics.html#print-root-of-daur2-project"><i class="fa fa-check"></i>Print root of DAUR2 project</a></li>
<li class="chapter" data-level="" data-path="guerilla-analytics.html"><a href="guerilla-analytics.html#root-of-daur2-project"><i class="fa fa-check"></i>Root of DAUR2 project</a></li>
<li class="chapter" data-level="" data-path="guerilla-analytics.html"><a href="guerilla-analytics.html#print-root-of-workflows-portfolio-project"><i class="fa fa-check"></i>Print root of workflows portfolio project</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="portfolio-assignment-1.1-c-elegans-plate.html"><a href="portfolio-assignment-1.1-c-elegans-plate.html"><i class="fa fa-check"></i><b>4</b> Portfolio assignment 1.1 C Elegans Plate</a>
<ul>
<li class="chapter" data-level="" data-path="portfolio-assignment-1.1-c-elegans-plate.html"><a href="portfolio-assignment-1.1-c-elegans-plate.html#anything-peculiar"><i class="fa fa-check"></i>Anything peculiar?</a></li>
<li class="chapter" data-level="" data-path="portfolio-assignment-1.1-c-elegans-plate.html"><a href="portfolio-assignment-1.1-c-elegans-plate.html#read-in-excel-file-in-r"><i class="fa fa-check"></i>Read in excel file in R</a></li>
<li class="chapter" data-level="" data-path="portfolio-assignment-1.1-c-elegans-plate.html"><a href="portfolio-assignment-1.1-c-elegans-plate.html#inspect-the-data-types-of-the-following-columns"><i class="fa fa-check"></i>Inspect the data types of the following columns</a></li>
<li class="chapter" data-level="" data-path="portfolio-assignment-1.1-c-elegans-plate.html"><a href="portfolio-assignment-1.1-c-elegans-plate.html#create-a-graph-displaying-a-scatterplot"><i class="fa fa-check"></i>Create a graph displaying a scatterplot</a></li>
<li class="chapter" data-level="" data-path="portfolio-assignment-1.1-c-elegans-plate.html"><a href="portfolio-assignment-1.1-c-elegans-plate.html#make-character-into-numeric"><i class="fa fa-check"></i>Make character into numeric</a></li>
<li class="chapter" data-level="" data-path="portfolio-assignment-1.1-c-elegans-plate.html"><a href="portfolio-assignment-1.1-c-elegans-plate.html#further-analysis"><i class="fa fa-check"></i>Further analysis</a></li>
<li class="chapter" data-level="" data-path="portfolio-assignment-1.1-c-elegans-plate.html"><a href="portfolio-assignment-1.1-c-elegans-plate.html#normalize-the-data-to-the-negative-control"><i class="fa fa-check"></i>normalize the data to the negative control</a></li>
<li class="chapter" data-level="" data-path="portfolio-assignment-1.1-c-elegans-plate.html"><a href="portfolio-assignment-1.1-c-elegans-plate.html#why-normalize"><i class="fa fa-check"></i>Why normalize?</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="open-peer-review.html"><a href="open-peer-review.html"><i class="fa fa-check"></i><b>5</b> Open Peer Review</a>
<ul>
<li class="chapter" data-level="" data-path="open-peer-review.html"><a href="open-peer-review.html#initial-peer-review-criteria-listing"><i class="fa fa-check"></i>Initial peer review criteria listing</a>
<ul>
<li class="chapter" data-level="" data-path="open-peer-review.html"><a href="open-peer-review.html#criteria-and-scoring-initial-peer-review"><i class="fa fa-check"></i>Criteria and scoring initial peer review</a></li>
<li class="chapter" data-level="" data-path="open-peer-review.html"><a href="open-peer-review.html#findings-initial-peer-review"><i class="fa fa-check"></i>Findings initial peer review</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="open-peer-review.html"><a href="open-peer-review.html#the-second-peer-review-with-code-and-data"><i class="fa fa-check"></i>The second peer review with code and data</a>
<ul>
<li class="chapter" data-level="5.0.1" data-path="open-peer-review.html"><a href="open-peer-review.html#osf-paper"><i class="fa fa-check"></i><b>5.0.1</b> OSF paper</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="the-easyread-r-package.html"><a href="the-easyread-r-package.html"><i class="fa fa-check"></i><b>6</b> The easyRead R package</a></li>
<li class="chapter" data-level="7" data-path="project-introduction-pedriactic-cancer.html"><a href="project-introduction-pedriactic-cancer.html"><i class="fa fa-check"></i><b>7</b> Project introduction: Pedriactic Cancer</a>
<ul>
<li class="chapter" data-level="" data-path="project-introduction-pedriactic-cancer.html"><a href="project-introduction-pedriactic-cancer.html#cfdna"><i class="fa fa-check"></i>cfDNA</a></li>
<li class="chapter" data-level="" data-path="project-introduction-pedriactic-cancer.html"><a href="project-introduction-pedriactic-cancer.html#r-shiny-app"><i class="fa fa-check"></i>R shiny app</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="parmaters-showcased-on-ecdc-results.html"><a href="parmaters-showcased-on-ecdc-results.html"><i class="fa fa-check"></i><b>8</b> Parmaters showcased on ECDC results</a>
<ul>
<li class="chapter" data-level="" data-path="parmaters-showcased-on-ecdc-results.html"><a href="parmaters-showcased-on-ecdc-results.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The portfolio of Nicky Klaver</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="free-space-assignment" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Free space assignment<a href="free-space-assignment.html#free-space-assignment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="introduction-single-cell-rna-sequencing" class="section level2 unnumbered hasAnchor">
<h2>Introduction single cell RNA sequencing<a href="free-space-assignment.html#introduction-single-cell-rna-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>At the start of my workflows course, I was presented by my teachers to perform a free space assignment. And as it sounds, it was completely free. It had to be Life Science related, and you had to learn something from it! The questions that are noted below where asked to me, and the answers I gave are shown under it.</p>
Where do I want to be in ±2 years?
<li>
I want to be able to execute a single cell sequencing dataset analysis and be able to use my git and github repocitories to make it reproducable for other people.
</li>
<br>
How am I doing now with respect to this goal?
<li>
I am learning how to use git and github, and I am thus making my work more reproducible.
</li>
<br>
What would be the next skill to learn?
<li>
I would need to learn how to analyse single cell sequencing, and also how to manipulate and communicate those results and visualize them as well.
</li>
<h2>
Planning:
</h2>
<li>
I have to find a single cell Seq. dataset and I have to inspect if it is something that I can work with.
</li>
<li>
I have to make the data tidy, then analyse the data with R and with bash.
</li>
<li>
I have to gather the results, and interpret these results into a graph of chart.
</li>
<li>
I have to put all this in a Rmarkdown which can be checked, and reproduced, only having
</li>
to insert the used data set.
<li>
I have to work via the workflow protocol, meaning with Git and Github.
</li>
<p><br></p>
<p>May 8th <br>
<i> 2-4 hours: </i> Finding the read count tables, and looking for associated code. Inspecting the data if it is usable.</p>
<p>May 14th <br>
<i> 6-8 hours: </i> How do I analyse this read count, and can I make an expression profile of these read counts.</p>
<p>May 15th <br>
<i> 3-4 hours: </i> Continue creating a expression profile, or if already finished, start identifying the type of cell.</p>
<p>May 21st <br>
<i> 6-7 hours: </i> Continue identifying the type of cell that is expressed by its expression profile.</p>
<p>May 26th <br>
<i> 2 hours: </i> Finish off the free space assignment, make sure everything is nice and the layout sits nice of the portfolio.</p>
<div id="start-of-the-free-time-project" class="section level3 unnumbered hasAnchor">
<h3>Start of the free time project:<a href="free-space-assignment.html#start-of-the-free-time-project" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>I started looking for YouTube videos that could give me some insights on how to perform Single Cell RNA Sequencing for some background information. It started with the following video’s (same creator, its a series, but not everything was as need so I filtered a little bit:</p>
<p>(Ep. 1) <span class="citation">(<a href="#ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023">SIB - Swiss Institute of Bioinformatics 2023f</a>)</span></p>
<p>(Ep. 3) <span class="citation">(<a href="#ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023e">SIB - Swiss Institute of Bioinformatics 2023a</a>)</span></p>
<p>(Ep. 4 halverwege gestart, daar was uitleg over wat je in R kon doen met bijvoorbeeld een PCA analysis, en met de package Seurat wat de functies daar van waren.) <span class="citation">(<a href="#ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023a">SIB - Swiss Institute of Bioinformatics 2023e</a>)</span></p>
<p>(Ep. 6 clustering) <span class="citation">(<a href="#ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023c">SIB - Swiss Institute of Bioinformatics 2023c</a>)</span></p>
<p>(Ep. 7 Cell type anotation) <span class="citation">(<a href="#ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023d">SIB - Swiss Institute of Bioinformatics 2023b</a>)</span></p>
<p>(Ep. 8 DGE analysis) <span class="citation">(<a href="#ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023b">SIB - Swiss Institute of Bioinformatics 2023d</a>)</span></p>
<hr />
<p>I gained information about what the workflow consists of, some different packages and also what some different analysis were that I could perform on the data set once I found a count table / count matrix.</p>
<p>Then I proceeded to watch the following <a href="https://www.youtube.com/watch?v=xbX49h7BiUU">video</a>.
From this video I gained information about meaning of some words, and again packages that I could use for analysis of Single Cell RNA Sequences</p>
</div>
<div id="meaning-of-words" class="section level3 unnumbered hasAnchor">
<h3>Meaning of words<a href="free-space-assignment.html#meaning-of-words" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<li>
UMI = Unique Molecular Identifiers, molecular tags that can be applied to detect and quantify the unique transcripts.
</li>
<li>
Features = genes
</li>
<li>
Bar codes = Single-cell sequencing experiments use short DNA bar code ‘tags’ to identify reads that originate from the same cell.
</li>
<li>
doublets = when two cells are encapsulated into one reaction volume.
</li>
</div>
<div id="r-packages-for-single-cell-rna-sequencing" class="section level3 unnumbered hasAnchor">
<h3>R packages for Single cell RNA sequencing<a href="free-space-assignment.html#r-packages-for-single-cell-rna-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<li>
SingleCellExperiment
</li>
<li>
Seurat
</li>
<li>
monocle3
</li>
<li>
Scater
</li>
<hr />
<p>After the YouTube video I went to search on her github page for some packages or data sets that I could use, which could or could not have included code and a count table.</p>
<p>Some time had past and I found a promising repository, and upon reading the README file is discovered some insights about also coursed and other things that I could follow to perform a Single Cell RNA Sequence:
<span class="citation">(<a href="#ref-AwesomesinglecellREADMEMd"><span>“Awesome-Single-Cell/<span>README</span>.md at Master <span><span class="math inline">\(\cdot\)</span></span> Seandavi/Awesome-Single-Cell”</span> n.d.</a>)</span></p>
<p>The cited course emerged in my hands, and I was eager to see what it could do for me <span class="citation">(<a href="#ref-CellgeniScRNASeq2024"><span>“Cellgeni/<span class="nocase">scRNA</span>.seq.course”</span> 2024</a>)</span>.
I was snooping around in all the files, and when looking at the README file I saw that there was a Webpage, and also a shared Data folder set! This of course is perfect for me, because I was needed to find a data set, with a read count table so that I could perform the analysis.</p>
<p>When looking at the website I saw the (presumably) read count table, it was named GSE149938_umi_matrix.csv.gz. This file was put into the raw data folder, and a copy was placed into the data folder.</p>
<p>I was searching for any sign of the Umi matrix being in correspondence with the read count table, and I came across the following website that said:</p>
<hr />
<p><em>“Sequencing data from single-cell RNA-seq experiments must be converted into a matrix of expression values. This is usually a count matrix containing the number of reads mapped to each gene (row) in each cell (column). Alternatively, the counts may be that of the number of unique molecular identifiers (UMIs); these are interpreted in the same manner as read counts but are less affected by PCR artifacts during library preparation”</em> <span class="citation"><em>Chapter 3 <span class="nocase">Getting scRNA-seq</span> Datasets <span></span> <span>Introduction</span> to <span>Single-Cell Analysis</span> with <span>Bioconductor</span></em> (<a href="#ref-ChapterGettingScRNAseq">n.d.</a>)</span></p>
<hr />
</div>
<div id="revealing-the-umi-matrix" class="section level3 unnumbered hasAnchor">
<h3>Revealing the Umi Matrix<a href="free-space-assignment.html#revealing-the-umi-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The UMI (Unique Molecular Identifier) matrix I downloaded was input into the Umi_matrix_NK object for further analysis. Downloaded from <span class="citation"><span>“S3 <span>Browser</span> <span></span> Singlecellcoursedata/”</span> (<a href="#ref-S3BrowserSinglecellcoursedata">n.d.</a>)</span>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="free-space-assignment.html#cb1-1" tabindex="-1"></a>Umi_matrix_NK <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb1-2"><a href="free-space-assignment.html#cb1-2" tabindex="-1"></a>  <span class="st">&quot;data&quot;</span>,</span>
<span id="cb1-3"><a href="free-space-assignment.html#cb1-3" tabindex="-1"></a>  <span class="st">&quot;GSE149938_umi_matrix.csv.gz&quot;</span>))</span></code></pre></div>
<p><br>
I found that there were multiple ways to import matrix data into R, one example I came up with myself was the above version. After further notice, and research, I found out this might not have been the best way.
<br>
I had read on the bioconducter website that I could best use read.delim or use the package <code>scuttle</code> and then use the function <code>readSparseCounts</code> because of the non zero-values not being taken into consideration. When looking at the Umi_matrix_NK object, I saw a lot of zero values, so I thought this could come in handy.
<br>
I performed the following commands:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="free-space-assignment.html#cb2-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb2-2"><a href="free-space-assignment.html#cb2-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb2-3"><a href="free-space-assignment.html#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="free-space-assignment.html#cb2-4" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;scuttle&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb2-5"><a href="free-space-assignment.html#cb2-5" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;scuttle&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="free-space-assignment.html#cb3-1" tabindex="-1"></a><span class="fu">library</span>(scuttle)</span>
<span id="cb3-2"><a href="free-space-assignment.html#cb3-2" tabindex="-1"></a>sparse_matrix_NK <span class="ot">&lt;-</span> <span class="fu">readSparseCounts</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;GSE149938_umi_matrix.csv.gz&quot;</span>))</span>
<span id="cb3-3"><a href="free-space-assignment.html#cb3-3" tabindex="-1"></a><span class="fu">dim</span>(sparse_matrix_NK) <span class="co"># check the dimensions</span></span></code></pre></div>
<p><br>
But when trying out the code I got the following error message:
<br></p>
<hr />
<p><em>Error in as(do.call(cbind, current[cell.cols]), “dgCMatrix”) : no method or default for coercing “NULL” to “dgCMatrix”</em></p>
<hr />
<p>And when you thought things couldn’t get any worse, git and github decided to not work as well. This because of the fact that on the Umi_matrix website, there also was another file called matrix, which I downloaded onto my computer.</p>
<p>I had put this into my raw_data folder, and copied it into my data folder. When I tried to push my changes to github, it gave me an error. I kinda ignored this error for another 2 commits before I checked what it said. The file was too big, it didn’t upload. This of course was a problem, but I found out I didn’t need this (300 MB) matrix file (I also had it twice so make that 600 MB). I tried a lot of different things, but it just did not work. I tried in the terminal, I tried in the git section in the top right. Nothing was working for me. In the terminal I looked for the git help page, and that did not do anything for me like a quick fix. It looked like I would be stuck for a little while.</p>
<p>Some awful time later on google and github forums I found the following solution (Hallelujah):
<em>The answer is simple: use the git reset –soft HEAD~1 command. This command will undo your last commit, but it will keep your changes and your staging intact.</em></p>
<p>So I tried, and nothing worked, so I typed it in again, and again, and again, and then I even changed it to HEAD~3 to make it every 3 commits. Then all of a sudden it all came back, my last (I think) 10 commits onto my screen. I repushed, and it stopped me in my tracks. I was thinking, when will this end. Then I did:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="free-space-assignment.html#cb4-1" tabindex="-1"></a><span class="fu">git</span> pull</span>
<span id="cb4-2"><a href="free-space-assignment.html#cb4-2" tabindex="-1"></a><span class="fu">git</span> add <span class="at">--all</span></span>
<span id="cb4-3"><a href="free-space-assignment.html#cb4-3" tabindex="-1"></a><span class="fu">git</span> push</span></code></pre></div>
<p>And now finally, after all this work. I was back to only my original problem. The <code>readSparseCounts</code> problem.</p>
<p>I went back to the <code>readSparseCounts</code> and saw that it did not work, so I went to look for how to load in UMI counts. I looked at the same website from the <code>readSparseCount</code> command and saw another method:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="free-space-assignment.html#cb5-1" tabindex="-1"></a>UMI_Matrix_NK <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;GSE149938_umi_matrix.csv.gz&quot;</span>)))</span>
<span id="cb5-2"><a href="free-space-assignment.html#cb5-2" tabindex="-1"></a><span class="fu">dim</span>(UMI_Matrix_NK) <span class="co"># shows the dimension of the matrix.</span></span></code></pre></div>
<p>Now after finally being able to load in the matrix, it was time to work on analyzing the matrix.</p>
<hr />
</div>
<div id="analyse-the-unique-molecular-identifier-umi-matrix" class="section level3 unnumbered hasAnchor">
<h3>Analyse the Unique Molecular Identifier (UMI) matrix<a href="free-space-assignment.html#analyse-the-unique-molecular-identifier-umi-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For the analysis for single cell RNA sequencing we shall use the SingleCellExperiment package that was noticed in 1 of the first video’s watched. Also we installed the Scater package for further analysis. And last but not least also the Seurat package was installed, because this was showcased in the video’s of the Swiss Institute of Bioinformatics.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="free-space-assignment.html#cb6-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb6-2"><a href="free-space-assignment.html#cb6-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb6-3"><a href="free-space-assignment.html#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="free-space-assignment.html#cb6-4" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;Seurat&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb6-5"><a href="free-space-assignment.html#cb6-5" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&#39;Seurat&#39;</span>)</span>
<span id="cb6-6"><a href="free-space-assignment.html#cb6-6" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb6-7"><a href="free-space-assignment.html#cb6-7" tabindex="-1"></a></span>
<span id="cb6-8"><a href="free-space-assignment.html#cb6-8" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;SingleCellExperiment&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;SingleCellExperiment&quot;</span>)</span>
<span id="cb6-9"><a href="free-space-assignment.html#cb6-9" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb6-10"><a href="free-space-assignment.html#cb6-10" tabindex="-1"></a></span>
<span id="cb6-11"><a href="free-space-assignment.html#cb6-11" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;scater&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb6-12"><a href="free-space-assignment.html#cb6-12" tabindex="-1"></a>   BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;scater&quot;</span>)</span>
<span id="cb6-13"><a href="free-space-assignment.html#cb6-13" tabindex="-1"></a><span class="fu">library</span>(scater)</span></code></pre></div>
<p>Now that we have installed and downloaded and loaded in the different packages, we can start by reading the documentation to see what functions it has. This was done by reading the manual online, as well as performing <code>?"insert package"</code> function in the terminal.</p>
</div>
<div id="analysis-overview" class="section level3 unnumbered hasAnchor">
<h3>Analysis overview<a href="free-space-assignment.html#analysis-overview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
First things first, we have to perform an analysis overview, which includes:
<li>
Cell filtering (quality control),
</li>
<li>
Normalization,
</li>
<li>
Feature selection,
</li>
<li>
Scaling,
</li>
<li>
Dimensionality reduction,
</li>
<li>
Clustering, annotation, DGE, etc.
</li>
<hr />
<div id="cell-filtering" class="section level5 unnumbered hasAnchor">
<h5>Cell filtering<a href="free-space-assignment.html#cell-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The following function was performed:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="free-space-assignment.html#cb7-1" tabindex="-1"></a><span class="co"># SingleCellExperiment function start of to make it an matrix with metadata of the rows and columns.</span></span>
<span id="cb7-2"><a href="free-space-assignment.html#cb7-2" tabindex="-1"></a>SCE_UMI_NK <span class="ot">&lt;-</span> <span class="fu">SingleCellExperiment</span>(<span class="fu">list</span>(<span class="at">counts=</span>UMI_Matrix_NK))</span>
<span id="cb7-3"><a href="free-space-assignment.html#cb7-3" tabindex="-1"></a>SCE_UMI_NK</span></code></pre></div>
<p>I came across another files set:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="free-space-assignment.html#cb8-1" tabindex="-1"></a>SCE_Reads_NK <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;raw_data&quot;</span>,<span class="st">&quot;reads.rds&quot;</span>))</span></code></pre></div>
</div>
</div>
<div id="scrna-course" class="section level3 unnumbered hasAnchor">
<h3>scRNA course!?!<a href="free-space-assignment.html#scrna-course" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>At this point in time I found that I was allowed to follow a course on scRNA, and that I should be able to perform it on a different data set.</p>
<p>The course was presented to my on the following <a href="https://sib-swiss.github.io/single-cell-training/">website</a>.</p>
<p>On this website I would be able to follow the course, and perform the analysis on a different data set as I went along, or so I thought.</p>
<p>There was another website where from I could get my data, that was right <a href="https://bioconductor.statistik.tu-dortmund.de/packages/3.9/workflows/vignettes/simpleSingleCell/inst/doc/reads.html#3_quality_control_on_the_cells">here</a> (or so I thought).</p>
<p>It seemed to me that all the data sets where different, they all just did not want to work with me. I did not give up hope, but it was kind of dissapointing because I had the feeling I could have gotten a lot further would’ve this dataset just worked with me like I wanted it to (# probably every scientist ever has said this).</p>
</div>
<div id="seurat-analysis" class="section level3 unnumbered hasAnchor">
<h3>Seurat analysis<a href="free-space-assignment.html#seurat-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After all the research, fideling, problems and etc. It was time for me to finally analyse. It had waited long enough. I found on the Seurat command list website the “normal” Seurat workflow, and I went with it. The Seurat website was found <a href="https://satijalab.org/seurat/articles/essential_commands.html">here</a>.</p>
<p>I started by loading in the UMI matrix that I first found. I loaded it into a normal object with the <code>here</code> function and <code>read.csv</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="free-space-assignment.html#cb9-1" tabindex="-1"></a>UMI_object <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb9-2"><a href="free-space-assignment.html#cb9-2" tabindex="-1"></a>  <span class="st">&quot;data&quot;</span>,</span>
<span id="cb9-3"><a href="free-space-assignment.html#cb9-3" tabindex="-1"></a>  <span class="st">&quot;GSE149938_umi_matrix.csv.gz&quot;</span>))</span></code></pre></div>
<p>After the data import I created a Seurtobject from the UMI matrix.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="free-space-assignment.html#cb10-1" tabindex="-1"></a>Seurat_object_UMI <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> UMI_object)</span></code></pre></div>
<p>After creating the Seurat object it said that it could not have underscores for the feautures but dashes, so it replaced these. Also the data is of class data frame, so it said “Coercing to dgCMatrix. This changing to a sparse matrix is I figured a problem, it would probably be more efficient.</p>
</div>
<div id="seurat-pre-processing" class="section level3 unnumbered hasAnchor">
<h3>Seurat pre-processing<a href="free-space-assignment.html#seurat-pre-processing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After the Seurat object was created I performed some pre-processing steps to the data like normalization, identify highly variable genes, and scaling the data.</p>
<p>The function <code>NormalizeData</code> is one function of the Seurat packages, and it performs a log-normalization on the count data given in the created seuratobject.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="free-space-assignment.html#cb11-1" tabindex="-1"></a>Seurat_object_UMI <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(Seurat_object_UMI) </span></code></pre></div>
<p>After this we want to identify high variable ‘features’ (genes), and we can use the function <code>FindVariableFeatures</code>. This function will identify features that are outliers on a ‘mean variability plot’.@FindVariableFeatures</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="free-space-assignment.html#cb12-1" tabindex="-1"></a>Seurat_object_UMI <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(Seurat_object_UMI)</span></code></pre></div>
<p>Another important preprocessing step is the scaling of the data. For this we use another Seurat function called <code>ScaleData</code>. This function scales and centers features in the dataset. <span class="citation"><span>“Scale and Center the Data. — <span>ScaleData</span>”</span> (<a href="#ref-ScaleCenterData">n.d.</a>)</span></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="free-space-assignment.html#cb13-1" tabindex="-1"></a>Seurat_object_UMI <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(Seurat_object_UMI)</span></code></pre></div>
</div>
<div id="seurat-dimensionality-reduction-and-clustering" class="section level3 unnumbered hasAnchor">
<h3>Seurat dimensionality reduction and clustering<a href="free-space-assignment.html#seurat-dimensionality-reduction-and-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As seen above here, Seurat is a package with a lot of functions to perform pre-processing on UMI matrixes. But it does not stop there. This is because there are also dimensionality reduction and clustering functions ready to be used.</p>
<p>The first one is the <code>RunPCA</code> function, this function will change the dimensions of the object and reduce them to a minimal.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="free-space-assignment.html#cb14-1" tabindex="-1"></a>Seurat_object_UMI <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(Seurat_object_UMI)</span></code></pre></div>
<p>The <code>RunPCA</code> function has as a default that it prints the first 5 Principle Components and shows the positives and the negatives (the features inside those dimensions, and the features outside those dimensions).</p>
<p>For the following analysis I will use 2 different objects and see which one is better. For this analysis we will look at the <code>RunUMAP</code> function. This runs the Uniform Manifold Approximation and Projection (UMAP) dimensional reduction technique.@RunUMAPRunUMAP</p>
<hr />
<p>The UMAP technique: “(UMAP) is a dimensionality reduction technique that constructs a high dimensional graph representation of the data then creates a low-dimensional graph to be as structurally similar as possible.@katherine-rittenbachWhatUMAP</p>
<hr />
<p>Because it could be wise to select the 10 most important principle components we could use this function with the argument <code>dims = 1:10</code>. But because I don’t want to lose the perhaps valuable information that is stored in the other dimensions I will also run the function with the argument <code>dims = 1:30</code> to be able to compare the data against each other and see if the dimensions make such a big difference. I will do this by performing the functions <code>RunUMAP</code> twice, and then follow up with the functions <code>FindNeighbours</code> and <code>FindClusters</code>, which are also functions of the <code>Seurat</code> Package. After this I will visualize both with the <code>DimPlot</code> function of (again) the <code>Seurat</code> package.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="free-space-assignment.html#cb15-1" tabindex="-1"></a><span class="co"># Perform the Seurat object UMAP on 10 dimensions</span></span>
<span id="cb15-2"><a href="free-space-assignment.html#cb15-2" tabindex="-1"></a>Seurat_object_UMI_10 <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(Seurat_object_UMI, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb15-3"><a href="free-space-assignment.html#cb15-3" tabindex="-1"></a>Seurat_object_UMI_10 <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(Seurat_object_UMI_10)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 19813
## Number of edges: 2776300
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.2565
## Number of communities: 2
## Elapsed time: 9 seconds</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="free-space-assignment.html#cb17-1" tabindex="-1"></a>Seurat_object_UMI_10 <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(Seurat_object_UMI, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb17-2"><a href="free-space-assignment.html#cb17-2" tabindex="-1"></a></span>
<span id="cb17-3"><a href="free-space-assignment.html#cb17-3" tabindex="-1"></a><span class="co"># Perform the Seurat object UMAP on 30 dimensions</span></span>
<span id="cb17-4"><a href="free-space-assignment.html#cb17-4" tabindex="-1"></a>Seurat_object_UMI_30 <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(Seurat_object_UMI, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb17-5"><a href="free-space-assignment.html#cb17-5" tabindex="-1"></a>Seurat_object_UMI_30 <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(Seurat_object_UMI_30)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 19813
## Number of edges: 3036394
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.2584
## Number of communities: 3
## Elapsed time: 11 seconds</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="free-space-assignment.html#cb19-1" tabindex="-1"></a>Seurat_object_UMI_30 <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(Seurat_object_UMI, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb19-2"><a href="free-space-assignment.html#cb19-2" tabindex="-1"></a></span>
<span id="cb19-3"><a href="free-space-assignment.html#cb19-3" tabindex="-1"></a><span class="co"># Visualize the dimensionality reduction and compare with gridarrange.</span></span>
<span id="cb19-4"><a href="free-space-assignment.html#cb19-4" tabindex="-1"></a>Seurat_10 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(Seurat_object_UMI_10, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span>
<span id="cb19-5"><a href="free-space-assignment.html#cb19-5" tabindex="-1"></a>Seurat_30 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(Seurat_object_UMI_10, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span>
<span id="cb19-6"><a href="free-space-assignment.html#cb19-6" tabindex="-1"></a><span class="fu">grid.arrange</span>(Seurat_10, Seurat_30, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/UMAP%2010%20and%2030-1.png" width="672" /></p>
<p>Looking at the 2 graphs, it looks like there is not real difference to using only the 10 dimensions, so it means I can further with that.</p>
</div>
<div id="seurat-expression-profile" class="section level3 unnumbered hasAnchor">
<h3>Seurat expression profile<a href="free-space-assignment.html#seurat-expression-profile" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now that pre-processing is finished, lets make it so that we can identify different cluster markers. The <code>Seurat</code> function <code>FindAllMarkers</code> finds markers (differentially expressed genes) for each of the identity classes in a dataset.@GeneExpressionMarkers</p>
<p>We want to identify each marker genes of each cluster.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="free-space-assignment.html#cb20-1" tabindex="-1"></a><span class="co"># First run again the function FindNeighbours and FindClusters</span></span>
<span id="cb20-2"><a href="free-space-assignment.html#cb20-2" tabindex="-1"></a>Seurat_object_UMI_10_c <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(Seurat_object_UMI, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb20-3"><a href="free-space-assignment.html#cb20-3" tabindex="-1"></a>Seurat_object_UMI_10_c <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(Seurat_object_UMI_10_c)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 19813
## Number of edges: 2776300
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.2565
## Number of communities: 2
## Elapsed time: 9 seconds</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="free-space-assignment.html#cb22-1" tabindex="-1"></a><span class="co"># Find all the markers</span></span>
<span id="cb22-2"><a href="free-space-assignment.html#cb22-2" tabindex="-1"></a>UMI_Markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(Seurat_object_UMI_10_c, </span>
<span id="cb22-3"><a href="free-space-assignment.html#cb22-3" tabindex="-1"></a>                              <span class="at">only.pos =</span> T, <span class="co"># Only return positive markers.</span></span>
<span id="cb22-4"><a href="free-space-assignment.html#cb22-4" tabindex="-1"></a>                              <span class="at">min.pct =</span> <span class="fl">0.25</span>, <span class="co"># Higher than 0.01 to speed up the process and find only genes that are not rarely expressed.</span></span>
<span id="cb22-5"><a href="free-space-assignment.html#cb22-5" tabindex="-1"></a>                              <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>) <span class="co"># Higher then 0.1 to speed up process, small change of missing weaker signals.</span></span></code></pre></div>
<p>After calculating the markers, we want to take out the top 10 markers for each cluster with the <code>top_n</code> function of the <code>dplyr</code> package.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="free-space-assignment.html#cb23-1" tabindex="-1"></a><span class="co"># Select top set of rows with top_n()</span></span>
<span id="cb23-2"><a href="free-space-assignment.html#cb23-2" tabindex="-1"></a>top_10markers_UMI <span class="ot">&lt;-</span> UMI_Markers <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">10</span>, <span class="co"># 10 rows picked</span></span>
<span id="cb23-3"><a href="free-space-assignment.html#cb23-3" tabindex="-1"></a>                                                                 <span class="at">wt =</span> avg_log2FC) <span class="co"># The variable used for ordering.</span></span>
<span id="cb23-4"><a href="free-space-assignment.html#cb23-4" tabindex="-1"></a></span>
<span id="cb23-5"><a href="free-space-assignment.html#cb23-5" tabindex="-1"></a><span class="co"># Print the top 10 genes</span></span>
<span id="cb23-6"><a href="free-space-assignment.html#cb23-6" tabindex="-1"></a><span class="fu">print</span>(top_10markers_UMI)</span></code></pre></div>
<pre><code>## # A tibble: 20 × 7
## # Groups:   cluster [2]
##        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene               
##        &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;              
##  1 0              14.3  0.442 0     0         0       ery-eBM7-L30-bar50 
##  2 1.36e-254      14.3  0.382 0     1.04e-250 0       LMPP-spBM2-L8-bar51
##  3 7.07e-250      15.4  0.377 0     5.40e-246 0       ery-eBM1-L9-bar36  
##  4 4.11e-249      13.9  0.376 0     3.14e-245 0       ery-eBM5-L27-bar93 
##  5 5.00e-247      14.0  0.373 0     3.82e-243 0       ery-eBM6-L22-bar41 
##  6 2.70e-244      13.9  0.37  0     2.07e-240 0       GMP-spBM2-L7-bar29 
##  7 2.79e-225      13.9  0.349 0     2.13e-221 0       HSC-spBM1-L6-bar68 
##  8 1.58e-218      14.0  0.341 0     1.21e-214 0       CMP-spBM1-L8-bar38 
##  9 3.65e-203      14.0  0.322 0     2.79e-199 0       ery-eBM5-L28-bar68 
## 10 2.77e-202      14.1  0.321 0     2.11e-198 0       GMP-spBM1-L7-bar74 
## 11 6.23e-212       3.59 0.013 0.36  4.76e-208 1       LMPP-spBM1-L1-bar49
## 12 1.00e-187       3.12 0.01  0.324 7.67e-184 1       BNK-spBM1-L2-bar49 
## 13 7.53e-165       2.83 0.004 0.282 5.76e-161 1       BNK-spBM1-L1-bar48 
## 14 1.20e-164       2.72 0     0.274 9.13e-161 1       cMOP-moBM3-L6-bar81
## 15 1.50e-162       3.34 0.005 0.28  1.15e-158 1       LMPP-spBM1-L1-bar63
## 16 1.64e-157       3.56 0.002 0.269 1.25e-153 1       MPP-spBM1-L2-bar21 
## 17 3.67e-155       2.90 0.001 0.263 2.80e-151 1       proB-bBM2-L15-bar35
## 18 1.25e-147       2.85 0.003 0.256 9.58e-144 1       HSC-spBM1-L1-bar20 
## 19 7.57e-144       3.13 0.003 0.251 5.78e-140 1       LMPP-spBM1-L4-bar11
## 20 6.21e-140       2.80 0.006 0.251 4.75e-136 1       MPP-spBM1-L4-bar61</code></pre>
<p>After now seeing the top 10 markers for both clusters, we can now visualize and compare these results. There are multiple ways to do this, but I will use the heatmap as learned in the DAUR2 lessons. Luckily, the <code>Seurat</code> package already has a function for this called <code>DoHeatmap</code>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="free-space-assignment.html#cb25-1" tabindex="-1"></a>UMI_Heatmap <span class="ot">&lt;-</span> <span class="fu">DoHeatmap</span>(Seurat_object_UMI_10_c, <span class="at">features =</span> top_10markers_UMI<span class="sc">$</span>gene)</span>
<span id="cb25-2"><a href="free-space-assignment.html#cb25-2" tabindex="-1"></a>UMI_Heatmap</span></code></pre></div>
<p><img src="_main_files/figure-html/heatmap-1.png" width="672" /></p>
<p>Sadly, after looking at the <code>UMI_Heatmap</code>, most of the features of cluster 1 were not taken into consideration because they were not found in the scale.data slot for the RNA assay.</p>
<p>Luckily (again), the <code>Seurat</code> package had another way of visualizing the features. This was either with the <code>DotPlot</code> function. The <code>DotPlot</code> is an intuitive way of visualizing how feature expression changes across different identity classes (clusters). The size of the dot encodes the percentage of cells within a class, while the color encodes the AverageExpression level across all cells within a class (blue is high). <span class="citation"><span>“Dot Plot Visualization — <span>DotPlot</span>”</span> (<a href="#ref-DotPlotVisualization">n.d.</a>)</span></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="free-space-assignment.html#cb26-1" tabindex="-1"></a><span class="co"># Perform the Dotplot on the UMI object</span></span>
<span id="cb26-2"><a href="free-space-assignment.html#cb26-2" tabindex="-1"></a>UMI_Dot <span class="ot">&lt;-</span> <span class="fu">DotPlot</span>(Seurat_object_UMI_10_c, <span class="at">features =</span> top_10markers_UMI)</span></code></pre></div>
<p>Sadly, also this command did not work, seeing as it gave the error:</p>
<hr />
<p><strong>Error in <code>levels&lt;-</code>(<code>*tmp*</code>, value = as.character(levels)) : factor level [54] is duplicated</strong></p>
<hr />
<p>The visualization did not really work for me. I tried to find the so called duplicates, but in the <code>top10markers_UMI</code> there were NO duplicates.</p>
<p>Even tho I did not get the visualization that I wanted. I did find a website <a href="http://xteam.xbio.top/CellMarker/search.jsp">here</a> where I could look up the feauture that I had found, so I tried that.</p>
<p>And that was for now the end of my free space assignment. I am eager to continue pursuing this route and hopefully finding a way to present that visualizations. Not IF but WHEN I get those visualizations, I will send them to my mentors.</p>

</div>
</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-AwesomesinglecellREADMEMd" class="csl-entry">
<span>“Awesome-Single-Cell/<span>README</span>.md at Master <span><span class="math inline">\(\cdot\)</span></span> Seandavi/Awesome-Single-Cell.”</span> n.d. <em>GitHub</em>. https://github.com/seandavi/awesome-single-cell/blob/master/README.md. Accessed May 14, 2024.
</div>
<div id="ref-CellgeniScRNASeq2024" class="csl-entry">
<span>“Cellgeni/<span class="nocase">scRNA</span>.seq.course.”</span> 2024. Cellular Genetics Informatics.
</div>
<div id="ref-ChapterGettingScRNAseq" class="csl-entry">
<em>Chapter 3 <span class="nocase">Getting scRNA-seq</span> Datasets <span></span> <span>Introduction</span> to <span>Single-Cell Analysis</span> with <span>Bioconductor</span></em>. n.d. Accessed May 14, 2024.
</div>
<div id="ref-DotPlotVisualization" class="csl-entry">
<span>“Dot Plot Visualization — <span>DotPlot</span>.”</span> n.d. https://satijalab.org/seurat/reference/dotplot. Accessed May 26, 2024.
</div>
<div id="ref-S3BrowserSinglecellcoursedata" class="csl-entry">
<span>“S3 <span>Browser</span> <span></span> Singlecellcoursedata/.”</span> n.d. https://singlecellcourse.cog.sanger.ac.uk/index.html?shared=data/. Accessed May 15, 2024.
</div>
<div id="ref-ScaleCenterData" class="csl-entry">
<span>“Scale and Center the Data. — <span>ScaleData</span>.”</span> n.d. https://satijalab.org/seurat/reference/scaledata. Accessed May 26, 2024.
</div>
<div id="ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023e" class="csl-entry">
SIB - Swiss Institute of Bioinformatics. 2023a. <span>“Single Cell Transcriptomics - <span>Analysis</span> Tools and Quality Control (3 of 10).”</span>
</div>
<div id="ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023d" class="csl-entry">
———. 2023b. <span>“Single Cell Transcriptomics - <span>Cell</span> Type Annotation (7 of 10).”</span>
</div>
<div id="ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023c" class="csl-entry">
———. 2023c. <span>“Single Cell Transcriptomics - <span>Clustering</span> (6 of 10).”</span>
</div>
<div id="ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023b" class="csl-entry">
———. 2023d. <span>“Single Cell Transcriptomics - <span>Differential</span> Gene Expression and <span>Enrichment</span> Analysis (8 of 10).”</span>
</div>
<div id="ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023a" class="csl-entry">
———. 2023e. <span>“Single Cell Transcriptomics - <span>Dimensionality</span> Reduction (4 of 10).”</span>
</div>
<div id="ref-sib-swissinstituteofbioinformaticsSingleCellTranscriptomics2023" class="csl-entry">
———. 2023f. <span>“Single Cell Transcriptomics - <span>Introduction</span> to Single Cell <span class="nocase">RNA-seq</span> (1 of 10).”</span>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cv.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="guerilla-analytics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
